lib/WGwrangler/GuiPlugin/WireguardAddPeerForm.pm

```diff
@@ -0,0 +1,245 @@
+            #         return "Your input does not look like an ip address";
+            #     }
+            #     return "";
+            # },
+            set    => {
+                required    => true,
+                placeholder => 'List of ip addresses, seperarated by comma and in CDIR notation'
+            },
+        },
+        {
+            key    => 'alias',
+            label  => trm('Alias'),
+            widget => 'text',
+            set    => {
+                placeholder => 'Unlike the name attribute, this must be unique'
+            }
+        },
+        {
+            key    => 'description',
+            label  => trm('Description'),
+            widget => 'textArea',
+            set    => {
+                placeholder => 'Some extra infos about this peer',
+            }
+        },
+    ];
+};
+
+has actionCfg => sub {
+    my $self = shift;
+
+    my $handler = sub {
+        my $self = shift;
+        my $args = shift;
+
+        my $interface = $args->{interface};
+        my $pub_key = $args->{'public-key'};
+        my $name = $args->{'name'};
+        my $desc = $args->{'description'};
+        my $ips = $args->{'allowed-ips'};
+        my $alias = $args->{'alias'};
+        $self->app->wireguardModel->add_peer($interface, $name, $ips, $pub_key, $alias, undef);
+        $self->app->wireguardModel->update_peer_data($interface, $pub_key, 'description', $desc);
+        $self->app->wireguardModel->commit_changes({});
+
+        return {
+            action => 'dataSaved'
+        };
+    };
+
+    return [
+        {
+            label         => trm('Add Peer'),
+            action        => 'submit',
+            key           => 'save',
+            actionHandler => $handler
+        }
+    ];
+};
+
+has grammar => sub {
+    my $self = shift;
+    $self->mergeGrammar(
+        $self->SUPER::grammar,
+        {
+            _doc  => "Tree Node Configuration",
+            _vars => [ qw(type) ],
+            type  => {
+                _doc => 'type of form to show: edit, add',
+                _re  => '(edit|add)'
+            },
+        },
+    );
+};
+
+sub generate_interface_label($self, $interface) {
+    my %interface_data = %{$self->app->wireguardModel->get_section_data($interface, $interface)};
+    my $iface_public_key = $self->app->wireguardModel->get_public_key($interface_data{'private-key'});
+    my $suggested_ips = $self->app->wireguardModel->suggest_ip($interface, 5);
+    my $label = "Interface: $interface \n"
+        . "Public-Key: $iface_public_key\n"
+        . "Address: $interface_data{'address'}\n"
+        . "FQDN: $interface_data{fqdn}:$interface_data{'listen-port'}\n"
+        . "Suggested-IPs: $suggested_ips";
+    return $label;
+}
+
+sub getAllFieldValues {
+    my $self = shift;
+    my $args = shift;
+    my $formData = shift;
+
+    my $data = $self->app->wireguardModel->gen_key_pair();
+
+    if ($formData->{currentFormData}{interface}) {
```

Comment:
is it safe to accept just any input from the outside at this point ? a bad actor could be feeding random data to the api